core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -211,6 +224,59 @@ private void runWithCompose(String cmd) {
                 .invoke();
     }
 
+    @SuppressWarnings("unchecked")
+    private static void validate(File composeFile) {
+        Yaml yaml = new Yaml();
+        try (FileInputStream fileInputStream = FileUtils.openInputStream(composeFile)) {
+            Object template = yaml.load(fileInputStream);
+            validate(template, composeFile.getAbsolutePath());
+        } catch (IOException e) {
+            log.warn("Failed to read YAML from {}", composeFile.getAbsolutePath(), e);
+        }
+    }
+
+    @VisibleForTesting
+    static void validate(Object template, String identifier) {
+        if (!(template instanceof Map)) {
+            return;
+        }
+
+        Map<String, ?> map = (Map<String, ?>) template;
+
+        final Map<String, ?> servicesMap;
+        if (map.containsKey("version")) {
+            if (!map.containsKey("services")) {
+                log.debug("Compose file {} has an unknown format: 'version' is set but 'services' is not defined", identifier);
+                return;
+            }
+            Object services = map.get("services");
+            if (!(services instanceof Map)) {
+                log.debug("Compose file {} has an unknown format: 'services' is not Map", identifier);
+                return;
+            }
+
+            servicesMap = (Map<String, ?>) services;
+        } else {
+            servicesMap = map;
+        }
+
+        for (Map.Entry<String, ?> entry : servicesMap.entrySet()) {
+            String serviceName = entry.getKey();
+            Object serviceDefinition = entry.getValue();
+            if (!(serviceDefinition instanceof Map)) {
+                break;
```

Comment:
Do we `break` rather than `continue` just to abort validation entirely?

Reply:
Will do 👍